pyclam/manifold.py

```diff
@@ -1181,15 +1174,56 @@ def clear_cache(self):
     def depth(self) -> int:
         return len(self.layers) - 1
 
+    def get_dot_file_string(self, graph_name):
+        """
+        checks over every cluster in the manifold, finding their parent edges and all edges between two clusters on the
+        same level, and then writes the information for each cluster and each edge onto a string that can be written
+        on a dot file
+        """
+        # starting the graph string with the graph name (an inputted value), and the root cluster
+        dot_file_string = 'graph ' + str(graph_name) + '{ \n' +\
+                          f'	edge[style=solid, penwidth="5",labeldistance="10"]\n' \
+                          f'	{self.root}[label="NODE: {self.root.name}\\nLFD: ' \
+                          f'{round(self.root.local_fractal_dimension, 3)}", ' \
+                          f'color="{self.root.color}", style=filled]\n'
+
+        # an array of clusters that were already written to the dot file
+        visited_clusters = []
+
+        # reading each layer of the manifold after the root
+        for i in range(1, self.depth + 1):
+            layer = self.layers[i]
+            edges = layer.build_edges().edges
+            for cluster in layer:
+                if cluster not in visited_clusters:
+
+                    # writing the current cluster, if it hasn't been visited already, to the string, and then writing
+                    # the edge between the cluster and its parent
+                    dot_file_string += f'	{cluster}[label="NODE: {cluster.name}\\nLFD: ' \
```

Comment:
I believe this loop could be a lot faster. Let's do a zoom call about the thought process here